### PR TITLE
refactor(compatibility): refactor stripe compatibility routes using `web::resource`

### LINKED ISSUE #961: [FEATURE] Refactor Stripe compatibility routes
### Feature Description

Use `web::resource` to specify the paths in [`/crates/router/src/compatibility/stripe/app.rs`](https://github.com/juspay/hyperswitch/blob/main/crates/router/src/compatibility/stripe/app.rs)

### Possible Implementation

Refer this file https://github.com/juspay/hyperswitch/blob/main/crates/router/src/routes/app.rs

### Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

None

### PR DESCRIPTION
**Pull request by nachiketkanore**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- fixes #961 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

- run cargo check on the entire project
- format the code

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/compatibility/stripe/app.rs [hunk 1]
```diff
@@ -9,12 +9,21 @@ impl PaymentIntents {
...
-            .service(payment_intents_update)
-            .service(payment_intents_confirm)
-            .service(payment_intents_capture)
+            .service(
+                web::resource("/sync")
+                    .route(web::post().to(payment_intents_retrieve_with_gateway_creds)),
+            )
+            .service(web::resource("/").route(web::post().to(payment_intents_create)))
+            .service(web::resource("/{payment_id}").route(web::get().to(payment_intents_retrieve)))
+            .service(web::resource("/{payment_id}").route(web::post().to(payment_intents_update)))
```
Comment (Abhicodes-crypto) [inline crates/router/src/compatibility/stripe/app.rs:18]:
```suggestion
            .service(
                    web::resource("/{payment_id}")
                        .route(web::get().to(payment_intents_retrieve))
                        .route(web::post().to(payment_intents_update)),
                )
```

File: crates/router/src/compatibility/stripe/app.rs [hunk 2]
```diff
@@ -9,12 +9,21 @@ impl PaymentIntents {
...
-            .service(payment_intents_create)
-            .service(payment_intents_retrieve)
-            .service(payment_intents_update)
-            .service(payment_intents_confirm)
-            .service(payment_intents_capture)
+            .service(
+                web::resource("/sync")
+                    .route(web::post().to(payment_intents_retrieve_with_gateway_creds)),
+            )
+            .service(web::resource("/").route(web::post().to(payment_intents_create)))
```
Comment (Abhicodes-crypto) [inline crates/router/src/compatibility/stripe/app.rs:16]: it should be `web::resource("/")`

File: crates/router/src/compatibility/stripe/app.rs (hunk 1 above)
Comment (Abhicodes-crypto) [reply crates/router/src/compatibility/stripe/app.rs:18]: Similar changes can be made for other paths wherever relevant.

File: crates/router/src/compatibility/stripe/app.rs [hunk 3]
```diff
@@ -24,10 +33,12 @@ impl SetupIntents {
     pub fn server(state: routes::AppState) -> Scope {
         web::scope("/setup_intents")
             .app_data(web::Data::new(state))
-            .service(setup_intents_create)
-            .service(setup_intents_retrieve)
-            .service(setup_intents_update)
-            .service(setup_intents_confirm)
+            .service(web::resource("/").route(web::post().to(setup_intents_create)))
```
Comment (Abhicodes-crypto) [inline crates/router/src/compatibility/stripe/app.rs:36]: It should be `web::resource("")`

File: crates/router/src/compatibility/stripe/app.rs [hunk 4]
```diff
@@ -37,10 +48,12 @@ impl Refunds {
...
             .app_data(web::Data::new(config))
-            .service(refund_create)
-            .service(refund_retrieve)
-            .service(refund_update)
-            .service(refund_retrieve_with_gateway_creds)
+            .service(web::resource("/").route(web::post().to(refund_create)))
+            .service(web::resource("/{refund_id}").route(web::get().to(refund_retrieve)))
+            .service(web::resource("/{refund_id}").route(web::post().to(refund_update)))
+            .service(
+                web::resource("/sync").route(web::post().to(refund_retrieve_with_gateway_creds)),
```
Comment (Abhicodes-crypto) [inline crates/router/src/compatibility/stripe/app.rs:55]: The `sync` route would match to `refund_update` which is above it so better to move this path to the top of refunds.

File: crates/router/src/compatibility/stripe/app.rs [hunk 5]
```diff
@@ -37,10 +48,12 @@ impl Refunds {
     pub fn server(config: routes::AppState) -> Scope {
         web::scope("/refunds")
             .app_data(web::Data::new(config))
-            .service(refund_create)
-            .service(refund_retrieve)
-            .service(refund_update)
-            .service(refund_retrieve_with_gateway_creds)
+            .service(web::resource("/").route(web::post().to(refund_create)))
```
Comment (Abhicodes-crypto) [inline crates/router/src/compatibility/stripe/app.rs:51]: It should be `web::resource("")`

File: crates/router/src/compatibility/stripe/app.rs [hunk 6]
```diff
@@ -24,10 +33,12 @@ impl SetupIntents {
     pub fn server(state: routes::AppState) -> Scope {
         web::scope("/setup_intents")
             .app_data(web::Data::new(state))
-            .service(setup_intents_create)
-            .service(setup_intents_retrieve)
-            .service(setup_intents_update)
-            .service(setup_intents_confirm)
+            .service(web::resource("/").route(web::post().to(setup_intents_create)))
+            .service(web::resource("/{setup_id}").route(web::get().to(setup_intents_retrieve)))
+            .service(web::resource("/{setup_id}").route(web::post().to(setup_intents_update)))
```
Comment (Abhicodes-crypto) [inline crates/router/src/compatibility/stripe/app.rs:38]:
```suggestion
      .service(
                    web::resource("/{setup_id}")
                        .route(web::get().to(setup_intents_retrieve))
                        .route(web::post().to(setup_intents_update)),
                )
```

**Review by Abhicodes-crypto — CHANGES_REQUESTED** [review]

File: crates/router/src/compatibility/stripe/customers.rs [hunk 7]
```diff
@@ -13,7 +13,7 @@ use crate::{
 };
 
 #[instrument(skip_all)]
-#[post("")]
+// #[post("")]
```
Comment (SanchithHegde) [inline crates/router/src/compatibility/stripe/customers.rs:16]: Please remove these comments from each of the route handlers.

File: crates/router/src/compatibility/stripe/payment_intents.rs [hunk 8]
```diff
@@ -1,6 +1,8 @@
 pub mod types;
 
-use actix_web::{get, post, web, HttpRequest, HttpResponse};
+#[cfg(feature = "olap")]
+use actix_web::get;
+use actix_web::{post, web, HttpRequest, HttpResponse};
```
Comment (SanchithHegde) [inline crates/router/src/compatibility/stripe/payment_intents.rs:5]: Please ensure that there are no more route handlers that use the `get` and `post` macros, and remove their imports.

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/router/src/compatibility/stripe/app.rs [hunk 9]
```diff
@@ -9,12 +9,24 @@ impl PaymentIntents {
...
+                    .route(web::post().to(payment_intents_update)),
+            )
+            .service(
+                web::resource("/{payment_id}/confirm")
+                    .route(web::post().to(payment_intents_confirm)),
+            )
+            .service(
+                web::resource("/{payment_id}/capture")
+                    .route(web::post().to(payment_intents_capture)),
+            )
```
Comment (Abhicodes-crypto) [inline crates/router/src/compatibility/stripe/app.rs:29]: Please add payments list and payments cancel as well.

Comment (nachiketkanore) [reply crates/router/src/compatibility/stripe/app.rs:29]: the /list method is for #[cfg(feature = "olap")]
How do I conditionally add that service to that scope?

Comment (SanchithHegde) [reply crates/router/src/compatibility/stripe/app.rs:29]: You should be able to do something like this (refer to the surrounding code for more context):

https://github.com/juspay/hyperswitch/blob/22a5372481bbf854cffb8b683606cdf0644a5f54/crates/router/src/routes/app.rs#L86-L89

**Review by SanchithHegde — APPROVED** [review]

**Comment by nachiketkanore** [comment]
changes done

**Review by Abhicodes-crypto — APPROVED** [review]
LGTM! Thank you for the PR.

**Review by jarnura — APPROVED** [review]

**Pull request merged**